sys/net/gnrc/transport_layer/udp/gnrc_udp.c

```diff
@@ -130,9 +130,9 @@ static void _receive(gnrc_pktsnip_t *pkt)
     port = (uint32_t)byteorder_ntohs(hdr->dst_port);
 
     /* send payload to receivers */
-    if (!gnrc_netapi_dispatch_receive(GNRC_NETTYPE_UDP, port, pkt)) {
+    if (!gnrc_netapi_dispatch_receive(GNRC_NETTYPE_UDP, port, udp)) {
```

Comment:
Maybe it's just this line that did the hard fault.. need to test if the other lines are necessary